### PR TITLE
Add timeout to requests calls

### PR DESCRIPTION
**Pull request by pixeeai**
Many developers will be surprised to learn that `requests` library calls do not include timeouts by default. This means that an attempted request could hang indefinitely if no connection is established or if no data is received from the server. 

The [requests documentation](https://requests.readthedocs.io/en/latest/user/advanced/#timeouts) suggests that most calls should explicitly include a `timeout` parameter. This codemod adds a default timeout value in order to set an upper bound on connection times and ensure that requests connect or fail in a timely manner. This value also ensures the connection will timeout if the server does not respond with data within a reasonable amount of time. 

While timeout values will be application dependent, we believe that this codemod adds a reasonable default that serves as an appropriate ceiling for most situations. 

Our changes look like the following:
```diff
 import requests
 
- requests.get("http://example.com")
+ requests.get("http://example.com", timeout=60)
```

<details>
  <summary>More reading</summary>

  * [https://docs.python-requests.org/en/master/user/quickstart/#timeouts](https://docs.python-requests.org/en/master/user/quickstart/#timeouts)
</details>

Powered by: [pixeebot](https://docs.pixee.ai/) (codemod ID: [pixee:python/add-requests-timeouts](https://docs.pixee.ai/codemods/python/pixee_python_add-requests-timeouts)) 
I have additional improvements ready for this repo! If you want to see them, install our plugin (Github Marketplace [download](https://github.com/apps/pixeebot)) & leave the comment:
```
@pixeebot next
```
... and I will open a new PR right away!

🧚🤖  Powered by Pixeebot

### CODE REVIEW

**Comment by ayaka14732** [comment]
Thanks! The PR looks good to me, but you need to sign the CLA first.

**Review by ayaka14732 — APPROVED** [review]
Thanks! The PR looks good to me, but you need to sign the CLA first.

File: jax/_src/clusters/cloud_tpu_cluster.py
Comment (yashk2810) [inline jax/_src/clusters/cloud_tpu_cluster.py:47]: Why default on 60? It would be nice to wait for @skye's approval on this.

Comment (skye) [reply jax/_src/clusters/cloud_tpu_cluster.py:47]: I think the author is a robot lol. 60 is probably too long but also fine (I think this usually fails or returns very quickly).

Comment (pixeeai) [reply jax/_src/clusters/cloud_tpu_cluster.py:47]: Hey - feel free to reference this line in the PR description for more clarity: ⬇️⬇️

"While timeout values will be application dependent, we believe that this codemod adds a reasonable default that serves as an appropriate ceiling for most situations."

We default to 60 seconds because that should be sufficient for most situations. Hope this helps!

Thanks,
Zach

**Comment by skye** [comment]
Are you able to sign the CLA?

**Comment by pixeeai** [comment]
> Are you able to sign the CLA?

Yes, I believe I got the issue resolved.

**Review by skye — APPROVED** [review]

**Comment by skye** [comment]
Thanks!

**Comment by pixeeai** [comment]
> Thanks!

No problem. Glad you guys liked the recommendation from @[pixeebot](https://github.com/pixee). We'd love it if you would install our free bot on GitHub's marketplace and provide us with any feedback. If interested, here is the [link](https://github.com/apps/pixeebot).

Thanks!

**Pull request merged**
